stock_request/models/stock_request_abstract.py

```diff
@@ -37,12 +37,17 @@ def _compute_product_qty(self):
 
     name = fields.Char("Name", copy=False, required=True, readonly=True, default="/")
     warehouse_id = fields.Many2one(
-        "stock.warehouse", "Warehouse", ondelete="cascade", required=True
+        comodel_name="stock.warehouse",
+        string="Warehouse",
+        check_company=True,
+        ondelete="cascade",
+        required=True,
     )
     location_id = fields.Many2one(
-        "stock.location",
-        "Location",
-        domain=[("usage", "in", ["internal", "transit"])],
+        comodel_name="stock.location",
+        string="Location",
+        domain="not allow_virtual_location and "
+        "[('usage', 'in', ['internal', 'transit'])] or [(1, '=', 1)]",
```

Comment:
Not sure if it should be empty domain instead:
```suggestion
        "[('usage', 'in', ['internal', 'transit'])] or []",
```